app/controllers/sites_controller.rb

```diff
@@ -226,7 +175,7 @@ def search_alert_site_by_collection collection_id, params
     search.exclude_id params[:exclude_id].to_i if params[:exclude_id].present?
     search.full_text_search params[:search] if params[:search].present?
     search.alerted_search params[:_alert] if params[:_alert].present?
-    search.my_site_search current_user.id unless current_user.can_view_other? collection_id
+    search.my_site_search current_user.id if current_user && !current_user.can_view_other?(collection_id)
```

Comment:
Seem searching my site is contrast with condition, I can search only when I can't view the other otherwise I can't